src/umleditor/mvc_model/custom_exceptions.py

```diff
@@ -76,6 +76,32 @@ class RelationDoesNotExistError(Error):
         def __init__(self, source, destination):
             super().__init__(f"Relation between '{source} -> {destination}' does not exist.")
 
+    class InvalidRelationTypeError(Error):
+        """
+        Exception raised when the relation being added has no types.
+        
+        Args:
+            source (Entity): The source of the relation that was being added.
+            destination (Entity): The destination of the relation that was
+                being added.
+
+        """
+        def __init__(self, source, destination):
+            super().__init__(f"Relation between '{source} -> {destination}' has invalid type.")
+
+    class InvalidRelationNewTypeError(Error):
+        """
+        Exception raised when the relation being added has invalid new type.
```

Comment:
for clarity, this means when you are changing the type of an existing relation and you try to change it to something invalid, correct?